live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/domain/model/out/event/UpdateNodeMetadataEvent.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * #%L
+ * Alfresco HX Insight Connector
+ * %%
+ * Copyright (C) 2023 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+
+package org.alfresco.hxi_connector.live_ingester.domain.model.out.event;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import lombok.NoArgsConstructor;
+import org.alfresco.hxi_connector.live_ingester.domain.model.out.NodeProperty;
+
+@NoArgsConstructor(staticName = "create")
+public class UpdateNodeMetadataEvent
+{
+
+    @JsonProperty("setProperties")
+    private final List<NodeProperty<?>> metadataPropertiesToSet = new ArrayList<>();
```

Comment:
I think this should be more like a `Map<String, Object>` since we can set each property to one value.

Reply:
yep, you're right, I'll change it to Map<String, NodeProperty >